pkg/minishift/openshift/service_test.go

```diff
@@ -0,0 +1,173 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openshift
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+
+	instanceState "github.com/minishift/minishift/pkg/minishift/config"
+	test "github.com/minishift/minishift/pkg/testing"
+	testdata "github.com/minishift/minishift/pkg/testing/testdata"
+)
+
+//
+type ByName []ServiceSpec
+
+func (a ByName) Len() int           { return len(a) }
+func (a ByName) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a ByName) Less(i, j int) bool { return a[i].Name < a[j].Name }
+
+func Test_getProject(t *testing.T) {
+	setup()
+	defer teardown()
+
+	fakeRunner := test.NewFakeRunner(t)
+	runner = fakeRunner.Runner
+
+	args := fmt.Sprintf("get projects --config=%s %s", systemKubeConfigPath, ProjectsCustomCol)
+	expectString := "NAME\nmyproject\ntest"
+	fakeRunner.ExpectAndReturn(args, expectString)
+
+	got, err := getProjects()
+
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
+
+	expected := strings.Split(expectString, "\n")[1:]
+	if !reflect.DeepEqual(got, expected) {
+		t.Fatalf("Expected: %+v, Got: %+v", expected, got)
+	}
+}
+
+func Test_getServiceSpec(t *testing.T) {
+	setup()
+	defer teardown()
+
+	service := "foo"
+	namespace := "bar"
+	https := false
+
+	fakeRunner := test.NewFakeRunner(t)
+	runner = fakeRunner.Runner
+
+	args := fmt.Sprintf("get projects --config=%s %s", systemKubeConfigPath, ProjectsCustomCol)
+	expectString := fmt.Sprintf("NAME\n%s", namespace)
+	fakeRunner.ExpectAndReturn(args, expectString)
+
+	args = fmt.Sprintf("get route/%s -n %s --config=%s %s", service, namespace, systemKubeConfigPath, URLCustomCol)
+	expectString = "URL\nfoo.bar.xip.io"
+	fakeRunner.ExpectAndReturn(args, expectString)
+
+	got, err := GetServiceSpec(service, namespace, https)
+
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
+
+	expected := "http://" + strings.Split(expectString, "\n")[1]
+
+	if https {
+		expected = "https://" + strings.Split(expectString, "\n")[1]
+	}
+
+	if got != expected {
+		t.Fatalf("Expected: %+v, Got: %+v", expected, got)
+	}
+
+}
+
+func Test_getServiceSpecs(t *testing.T) {
+	setup()
+	defer teardown()
+
+	namespace := "bar"
+
+	fakeRunner := test.NewFakeRunner(t)
+	runner = fakeRunner.Runner
+
+	args := fmt.Sprintf("get projects --config=%s %s", systemKubeConfigPath, ProjectsCustomCol)
+	expectString := fmt.Sprintf("NAME\n%s", namespace)
+	fakeRunner.ExpectAndReturn(args, expectString)
+
+	args = fmt.Sprintf("get svc -o json --config=%s -n %s", systemKubeConfigPath, namespace)
+	expectString = testdata.ServiceSpec
```

Comment:
nice, using some real test data. This way the whole JSON processing gets tested +1